packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -266,6 +274,49 @@ export interface FunctionProps extends FunctionOptions {
  * library.
  */
 export class Function extends FunctionBase {
+
+  /**
+   * Returns a `lambda.Version` which represents the current version of this
+   * Lambda function. A new version will be created every time the function's
+   * configuration changes.
+   *
+   * You can specify options for this version using the `currentVersionOptions`
+   * prop when initializing the `lambda.Function`.
+   */
+  public get currentVersion(): Version {
+    if (this._currentVersion) {
+      return this._currentVersion;
+    }
+
+    this._currentVersion = new Version(this, `CurrentVersion`, {
+      lambda: this,
+      description: `${this.node.path}.currentVersion (created by AWS CDK)`,
```

Comment:
We should not add a description. This can lead to very unfortunate data loss on behalf of the customer, and then if we decide to remove it, it also causes failed deployments.

The data loss is that if the customer put useful metadata in the function on $LATEST, publish-version with a description overrides that data. It's unintuitive and dangerous feature (but also useful for a specific subset of customers).

Reply:
Okay, you scared me... Removing :-)